utils/src/treev2/quickPickWizard/quickPickAzureResource/QuickPickAzureSubscriptionStep.ts

```diff
@@ -0,0 +1,42 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as vscode from 'vscode';
+import { AzureResourceQuickPickWizardContext } from '../../../../hostapi.v2';
+import { PickFilter } from '../common/PickFilter';
+import { GenericQuickPickOptions, GenericQuickPickStep, SkipIfOneQuickPickOptions } from '../GenericQuickPickStep';
+import { ResourceGroupsItem, SubscriptionItem } from './tempTypes';
+
+export class QuickPickAzureSubscriptionStep extends GenericQuickPickStep<AzureResourceQuickPickWizardContext, SkipIfOneQuickPickOptions> {
+    public constructor(tdp: vscode.TreeDataProvider<ResourceGroupsItem>, options?: GenericQuickPickOptions) {
+        super(tdp, {
+            ...options,
+            skipIfOne: true, // Subscription is always skip-if-one
+        });
+    }
+
+    protected readonly pickFilter = new AzureSubscriptionPickFilter();
+
+    protected override async promptInternal(wizardContext: AzureResourceQuickPickWizardContext): Promise<SubscriptionItem> {
+        const pickedSubscription = await super.promptInternal(wizardContext) as SubscriptionItem;
+
+        // TODO
+        wizardContext.subscription = pickedSubscription.subscription;
+
+        return pickedSubscription;
+    }
+}
+
+class AzureSubscriptionPickFilter implements PickFilter {
+    isDirectPick(_node: vscode.TreeItem): boolean {
+        // Subscription is never a direct pick
+        return false;
+    }
+
+    isIndirectPick(_node: vscode.TreeItem): boolean {
+        // All nodes at this level are always subscription nodes
+        return true;

```

Comment:
Should we have a different step for handling workspace _and_ Subscription nodes?

Reply:
We still have to work out how we want selecting from either a subscription or a workspace item to work. Once I'm further along with the "experience builder" we can go from there on how to tackle this.

Reply:
It's currently broken for all of the extensions that used to: Functions, Storage, and Databases.